File: frontend/src/types/analytics.ts
export interface ClubStats {
  total_members: number;
  active_members: number;
  sessions_this_week: number;
  sessions_this_month: number;
  most_active_member_name: string | null;
  top_exercise_name: string | null;
  total_sessions: number;
}

export interface E1RMDataPoint {
  date: string;
  e1rm: number;
  exercise_name: string;
  is_pr: boolean;
}

export interface VolumeDataPoint {
  date: string;
  volume: number;
  exercise_name: string;
}

export interface ProjectionResult {
  exercise_id: number;
  exercise_name: string;
  current_e1rm: number;
  projected_4wk: number;
  projected_8wk: number;
  r_squared: number;
  insufficient_data: boolean;
}

export interface PlateauResult {
  exercise_id: number;
  exercise_name: string;
  is_plateau: boolean;
  last_3_pct_change: number | null;
  sessions_analyzed: number;
}

export interface Recommendation {
  exercise_id: number;
  exercise_name: string;
  recommendation_type: string;
  message: string;
  severity: 'info' | 'warning' | 'success';
}

export interface NeedsAttentionMember {
  id: number;
  name: string;
  reasons: string[];
  days_since_last_session: number | null;
  last_session_date: string | null;
}

export interface WeeklyVolume {
  week_start: string;
  total_volume: number;
  session_count: number;
}

export interface TopPerformer {
  member_id: number;
  member_name: string;
  exercise_name: string;
  e1rm_gain_pct: number;
  current_e1rm: number;
}

File: backend/app/schemas/analytics.py
from datetime import date
from typing import Optional, List
from pydantic import BaseModel


class ClubStats(BaseModel):
    total_members: int
    active_members: int
    sessions_this_week: int
    sessions_this_month: int
    most_active_member_name: Optional[str]
    top_exercise_name: Optional[str]
    total_sessions: int


class E1RMDataPoint(BaseModel):
    date: date
    e1rm: float
    exercise_name: str
    is_pr: bool = False


class VolumeDataPoint(BaseModel):
    date: date
    volume: float
    exercise_name: str


class ProjectionResult(BaseModel):
    exercise_id: int
    exercise_name: str
    current_e1rm: float
    projected_4wk: float
    projected_8wk: float
    r_squared: float
    insufficient_data: bool = False


class PlateauResult(BaseModel):
    exercise_id: int
    exercise_name: str
    is_plateau: bool
    last_3_pct_change: Optional[float] = None
    sessions_analyzed: int = 0


class Recommendation(BaseModel):
    exercise_id: int
    exercise_name: str
    recommendation_type: str
    message: str
    severity: str = "info"  # info, warning, success


class NeedsAttentionMember(BaseModel):
    id: int
    name: str
    reasons: List[str]
    days_since_last_session: Optional[int]
    last_session_date: Optional[date]


class WeeklyVolume(BaseModel):
    week_start: date
    total_volume: float
    session_count: int


class TopPerformer(BaseModel):
    member_id: int
    member_name: str
    exercise_name: str
    e1rm_gain_pct: float
    current_e1rm: float


class TopRepGainer(BaseModel):
    member_id: int
    member_name: str
    exercise_name: str
    reps_gain_pct: float
    current_reps: int


class RepsDataPoint(BaseModel):
    date: date
    reps: int
    exercise_name: str
    is_pr: bool = False


class RepsProjectionResult(BaseModel):
    exercise_id: int
    exercise_name: str
    current_reps: int
    projected_4wk: float
    projected_8wk: float
    r_squared: float
    insufficient_data: bool = False
